lib/apiservers/engine/backends/container.go

```diff
@@ -584,12 +630,36 @@ func (c *Container) mapPorts(op portmap.Operation, hostconfig *containertypes.Ho
 				}
 			}
 
+			// check if we should actually unmap based on current mappings
```

Comment:
This whole code block that accesses `containerByPort` needs to be protected with a lock. Maybe even acquire the lock for the duration of the for loop.